components/plan-build-ps1/bin/hab-plan-build.ps1

```diff
@@ -1550,20 +1555,32 @@ $candidateTargetPaths = @(
     "$PLAN_CONTEXT\$pkg_target\plan.ps1",
     "$PLAN_CONTEXT\habitat\$pkg_target\plan.ps1"
 )
-foreach($place in $candidateTargetPaths) {
-    if(Test-Path $place) {
-        $paths += $place
-        $script:PLAN_CONTEXT = (Split-Path $place -Parent)
+foreach($path in $candidateTargetPaths) {
+    if(Test-Path $path) {
+        # Take note of the found plan file and keep iterating
+        # because later we want to know if we found more than one
+        $targetPaths += $path
+        $script:PLAN_CONTEXT = (Split-Path $path -Parent)
     }
 }
-if($paths.Count -eq 0) {
-    foreach($place in $candidatePaths) {
-        if(Test-Path $place) {
-            $paths += $place
-            $script:PLAN_CONTEXT = (Split-Path $place -Parent)
-        }
+
+foreach($path in $candidatePaths) {
+    if(Test-Path $path) {
+        # Take note of the found plan file and keep iterating
+        # because later we want to know if we found more than one
+        $paths += $path
+        $script:PLAN_CONTEXT = (Split-Path $path -Parent)
     }
 }
+
+if($paths.Count -gt 0 -and ($targetPaths.Count -gt 0)) {
+    Write-Warning "There is a plan.ps1 inside $pkg_target and outside as well. Using the plan in $pkg_target."
+    Write-Warning "It is advisable to either remove the plan that is outside $pkg_target"
+    Write-Warning "or move that plan to its own target folder if it is intended for a different target."
+}
+if($targetPaths.Count -gt 0) {
+    $paths = $targetPaths
```

Comment:
This operation coupled with the `if / throw` statements below is more confusing and error prone than it should be. Perhaps creating a `$finalPaths` variable would be good, since that makes things explicit, and keeps the `$paths` and `$targetPaths` variables single-purpose.